tests/test_preprocessing.py

```diff
@@ -146,3 +150,42 @@ def test_make_dataset_from_sgf(self):
                 -1
             )]
         self.assertEqualData(expected_data, recovered_data)
+
+
+    def test_rotate_pyfunc(self):
+        def test_symmetry(s, i, r1, r2):
+            # features
+            x = (r1[0][i] == symmetries.apply_symmetry_feat(sym, r2[0][i])).all()
+            # pi
+            pi = (r1[1][i] == symmetries.apply_symmetry_pi(sym, r2[1][i])).all()
+            # value
+            v = (r1[2][i] == r2[2][i]).all()
+            return x and pi and v
+
+        num_records = 200
+        raw_data = self.create_random_data(num_records)
+        tfexamples = list(map(preprocessing.make_tf_example, *zip(*raw_data)))
+
+        with tempfile.NamedTemporaryFile() as f:
+            preprocessing.write_tf_examples(f.name, tfexamples)
+            run_one = self.extract_data(
+                f.name, filter_amount=1, random_rotation=False)
+            run_two = self.extract_data(
+                f.name, filter_amount=1, random_rotation=True)
+
```

Comment:
This code will pass even if random_rotation does nothing; nowhere do you assert that run_one's data and run_two's data are actually different.  Just add an assert that run_one != run_two

Reply:
Below I verify that I find all eight rotations which wouldn't pass but it's easy to add.

Reply:
That's not quite what it does -- it just finds *a* rotation that matches, then it falls out of the loop.  If the identity rotation passed (because run_one == run_two), the loop would still be exited w/o failing the test.

Reply:
you're not quite verifying the behavior you think you're verifying, is what i'm sayin ;)

Reply:
Disagree, look at line 191

```
difference = set(syms) ^ set(symmetries.SYMMETRIES)
assert not difference, "Didn't find {}".format(difference)
```